pkg/controller/installation/core_controller.go

```diff
@@ -913,11 +925,37 @@ func (r *ReconcileInstallation) Reconcile(ctx context.Context, request reconcile
 		}
 	}
 
+	// nodeReporterMetricsPort is a port used in Enterprise to host internal metrics.
+	// Operator is responsible for creating a service which maps to that port.
+	// Here, we'll check the default felixconfiguration to see if the user is specifying
+	// a non-default port, and use that value if they are.
+	nodeReporterMetricsPort := defaultNodeReporterPort
+	if instance.Spec.Variant == operator.TigeraSecureEnterprise {
+		// Query the FelixConfiguration object. We'll use this to help configure felix.
+		felixConfiguration := &crdv1.FelixConfiguration{}
+		err = r.client.Get(ctx, types.NamespacedName{Name: "default"}, felixConfiguration)
+		if err != nil && !apierrors.IsNotFound(err) {
+			r.SetDegraded("Unable to read FelixConfiguration", err, reqLogger)
+			return reconcile.Result{}, err
+		}
+
+		// Determine the port to use for nodeReporter metrics.
+		if felixConfiguration.Spec.PrometheusReporterPort != nil {
+			nodeReporterMetricsPort = *felixConfiguration.Spec.PrometheusReporterPort
+		}
+
+		if nodeReporterMetricsPort == 0 {
+			err := errors.New("felixConfiguration prometheusReporterPort=0 not supported")
+			r.SetDegraded("", err, reqLogger)
```

Comment:
I think we need to have something in the 1st field because that is the Degraded "reason", I think saying "Invalid metrics port" would work and you can keep the error message as you have it.